lightning/src/util/test_utils.rs

```diff
@@ -238,35 +237,39 @@ pub struct TestPersister {
 impl TestPersister {
 	pub fn new() -> Self {
 		Self {
-			update_ret: Mutex::new(chain::ChannelMonitorUpdateStatus::Completed),
-			next_update_ret: Mutex::new(None),
+			update_rets: Mutex::new(VecDeque::new()),
 			chain_sync_monitor_persistences: Mutex::new(HashMap::new()),
 			offchain_monitor_updates: Mutex::new(HashMap::new()),
 		}
 	}
 
+	/// Clear the queue of update statuses and set the one we'll always return.
 	pub fn set_update_ret(&self, ret: chain::ChannelMonitorUpdateStatus) {
-		*self.update_ret.lock().unwrap() = ret;
+		let mut update_rets = self.update_rets.lock().unwrap();
+		update_rets.clear();
+		update_rets.push_front(ret);
 	}
 
-	pub fn set_next_update_ret(&self, next_ret: Option<chain::ChannelMonitorUpdateStatus>) {
-		*self.next_update_ret.lock().unwrap() = next_ret;
+	/// Queue an update status to return.
+	pub fn set_next_update_ret(&self, next_ret: chain::ChannelMonitorUpdateStatus) {
+		self.update_rets.lock().unwrap().push_back(next_ret);
 	}
 }
 impl<Signer: keysinterface::Sign> chainmonitor::Persist<Signer> for TestPersister {
 	fn persist_new_channel(&self, _funding_txo: OutPoint, _data: &channelmonitor::ChannelMonitor<Signer>, _id: MonitorUpdateId) -> chain::ChannelMonitorUpdateStatus {
-		let ret = self.update_ret.lock().unwrap().clone();
-		if let Some(next_ret) = self.next_update_ret.lock().unwrap().take() {
-			*self.update_ret.lock().unwrap() = next_ret;
-		}
-		ret
+		let mut update_rets = self.update_rets.lock().unwrap();
+		if update_rets.len() > 1 { return update_rets.pop_front().unwrap() }
+		else if update_rets.len() == 1 { return *update_rets.front().clone().unwrap() }
+		chain::ChannelMonitorUpdateStatus::Completed
```

Comment:
A little more idiomatic if this were `if { .. } else if { .. } else { .. }` without any explicit returns.